packages/ripple/src/runtime/internal/client/runtime.js

```diff
@@ -278,6 +279,89 @@ export function derived(fn, block) {
   };
 }
 
+/**
+ * @param {any} v
+ * @param {TrackOptions | undefined} o
+ * @param {Block} b
+ * @returns {Tracked | Derived | Tracked[]}
+ */
+export function track(v, o, b) {
+  var is_tracked = is_tracked_object(v);
+
+  if (o === undefined) {
+    if (is_tracked) {
+      return v;
+    }
+
+    if (typeof v === 'function') {
+      return derived(v, b);
+    }
+    return tracked(v, b);
+  }
+
+  if (is_tracked || typeof v !== "object" || v === null || is_array(v)) {
+    throw new TypeError('Invalid value: expected a non-tracked object');
+  }
+
+  var list;
+
+  if (
+    typeof o !== "object" ||
+    o === null ||
+    is_array(o) ||
+    !("split" in o) ||
+    !is_array(list = o.split) ||
+    list.length < 1
+  ) {
+    throw new TypeError(
+      'Invalid options: expected { split: (string|symbol|number)[] }'
+    );
+  }
+
+  /** @type {Tracked[]} */
+  var out = []
+  /** @type {Record<string|symbol, any>} */
+  var rest = {};
+  /** @type {Record<PropertyKey, PropertyDescriptor>} */
+  var descriptors = get_descriptors(v);
+  var props = Reflect.ownKeys(descriptors);
+  /** @type {Record<string|symbol, Tracked>} */
+  var done = {};
+
+  for (let i = 0, props_i = 0, key, t; i < list.length; i++) {
+    key = list[i];
+
+    if (done[key]) {
+      // in case of duplicate keys just return as requested
+      out[i] = done[key];
```

Comment:
This isn't needed. We can just re-apply duplicates, it's not going to cause issues